### PR TITLE
[Beta]: docs(cn): state-a-components-memory

### PR DESCRIPTION
**Pull request by zzwgh**

### CODE REVIEW

File: beta/src/pages/learn/state-a-components-memory.md [hunk 1]
```diff
@@ -1435,19 +1439,20 @@ export default function FeedbackForm() {
...
 
-### Remove unnecessary state
+### 移除不必要的 state
+
+当按钮被点击时，这个例子应该询问用户的名字，然后显示一个 alert 欢迎他们。你尝试使用 state 来保存名字，但由于某种原因，它始终显示“Hello, ！”。
 
-When the button is clicked, this example should ask for the user's name and then display an alert gretting them. You tried to use state to keep the name, but for some reason it always shows "Hello, !".
 
-To fix this code, remove the unnecessary state variable. (We will discuss about [why this didn't work](/learn/troubleshooting-state-updates#setting-state-does-not-update-variables) later.)
+要修复此代码，请删除不必要的 state 变量。（我们将在稍后讨论 [为什么上述代码不起作用](/learn/troubleshooting-state-updates#setting-state-does-not-update-variables)。）
```
Comment (x-cold) [inline beta/src/pages/learn/state-a-components-memory.md:1453]:
```suggestion
要修复此代码，请删除不必要的 state 变量。（我们将在稍后讨论[为什么上述代码不起作用](/learn/troubleshooting-state-updates#setting-state-does-not-update-variables)。）
```

File: beta/src/pages/learn/state-a-components-memory.md [hunk 2]
```diff
@@ -1474,7 +1479,7 @@ export default function FeedbackForm() {
 
 <Solution>
 
-Here is a fixed version that uses a regular `name` variable declared in the function that needs it:
+以下是一个使用了一般变量 `name` 的固定版本，这个变量声明于需要它的函数中。
```
Comment (x-cold) [inline beta/src/pages/learn/state-a-components-memory.md:1482]:
```suggestion
以下是一个使用了普通变量 `name` 的固定版本，这个变量声明于需要它的函数中。
```

File: beta/src/pages/learn/state-a-components-memory.md [hunk 3]
```diff
@@ -1497,7 +1502,7 @@ export default function FeedbackForm() {
 
 </Sandpack>
 
-A state variable is only necessary to keep information between re-renders of a component. Within a single event handler, a regular variable will do fine. Don't introduce state variables when a regular variable works well.
+State 变量仅用于在组件重渲染时保存信息。在单个事件处理函数中，一般变量就足够了。当一般变量运行良好时，不要引入 state 变量。
```
Comment (x-cold) [inline beta/src/pages/learn/state-a-components-memory.md:1505]:
```suggestion
State 变量仅用于在组件重渲染时保存信息。在单个事件处理函数中，普通变量就足够了。当普通变量运行良好时，不要引入 state 变量。
```

File: beta/src/pages/learn/state-a-components-memory.md [hunk 4]
```diff
@@ -1,25 +1,27 @@
 ---
-title: "State: A Component's Memory"
+title: "State：组件的记忆"
+translators:
+  - 7ooz
```
Comment (x-cold) [inline beta/src/pages/learn/state-a-components-memory.md:4]:
```suggestion
  - 7ooz
  - x-cold
```

File: beta/src/pages/learn/state-a-components-memory.md [hunk 5]
```diff
@@ -151,54 +153,55 @@ button {
...
-2. **Changes to local variables won't trigger renders.** React doesn't realize it needs to render the component again with the new data.
+1. **局部变量无法在多次渲染中持久保存。** 当 React 第二次渲染这个组件时，它会从头开始渲染——不考虑之前对局部变量的任何更改。
+2. **更改局部变量不会触发渲染。** React 没有意识到它需要使用新数据再次渲染组件。
 
-To update a component with new data, two things need to happen:
+要使用新数据更新组件，需要做两件事：
 
-1. **Retain** the data between renders.
-2. **Trigger** React to render the component with new data (re-rendering).
+1. **保留**渲染之间的数据。
```
Comment (KnowsCount) [inline beta/src/pages/learn/state-a-components-memory.md:163]:
```suggestion
1. **保留** 渲染之间的数据。
```

**Comment by awxiaoxian2020** [comment]
Do you consider applying code review or looking into it?

**Comment by zzwgh** [comment]
> Do you consider applying code review or looking into it?
> 
> Or you have done these changes?

last round of conversations are resolved. maybe need some more review on it.

File: beta/src/content/learn/state-a-components-memory.md [hunk 6]
```diff
@@ -331,53 +335,53 @@ button {
...
+### 遇见你的第一个 Hook {/*meet-your-first-hook*/}
 
-In React, `useState`, as well as any other function starting with "`use`", is called a Hook.
+在 React 中，`useState` 以及任何其他以“`use`”开头的函数都被称为 **Hook**。
 
-*Hooks* are special functions that are only available while React is [rendering](/learn/render-and-commit#step-1-trigger-a-render) (which we'll get into in more detail on the next page). They let you "hook into" different React features.
+Hooks 是特殊的函数，只在 React [渲染](/learn/render-and-commit#step-1-trigger-a-render)时有效（我们将在下一节详细介绍）。它们能让你 “hook” 到不同的 React 特性中去。
 
-State is just one of those features, but you will meet the other Hooks later.
+State 只是这些特性中的一个，你之后还会遇到其他 Hooks。
```
Comment (daochouwangu) [inline beta/src/content/learn/state-a-components-memory.md:344]:
```suggestion
State 只是这些特性中的一个，你之后还会遇到其他 Hook。
```

File: beta/src/content/learn/state-a-components-memory.md [hunk 7]
```diff
@@ -331,53 +335,53 @@ button {
...
 
 <Pitfall>
 
-**Hooks—functions starting with `use`—can only be called at the top level of your components or [your own Hooks.](/learn/reusing-logic-with-custom-hooks)** You can't call Hooks inside conditions, loops, or other nested functions. Hooks are functions, but it's helpful to think of them as unconditional declarations about your component's needs. You "use" React features at the top of your component similar to how you "import" modules at the top of your file.
+**Hooks——以 `use` 开头的函数——只能在组件或[自定义 Hooks](/learn/reusing-logic-with-custom-hooks) 的最顶层调用。** 你不能在条件语句、循环语句或其他嵌套函数内调用 Hooks。Hooks 是函数，但将它们视为关于组件需求的无条件声明会很有帮助。在组件顶部 “use” React 特性，类似于在文件顶部“导入”模块。
```
Comment (daochouwangu) [inline beta/src/content/learn/state-a-components-memory.md:348]:
```suggestion
**Hooks ——以 `use` 开头的函数——只能在组件或[自定义 Hook](/learn/reusing-logic-with-custom-hooks) 的最顶层调用。** 你不能在条件语句、循环语句或其他嵌套函数内调用 Hook。Hook 是函数，但将它们视为关于组件需求的无条件声明会很有帮助。在组件顶部 “use” React 特性，类似于在文件顶部“导入”模块。
```

File: beta/src/content/learn/state-a-components-memory.md [hunk 8]
```diff
@@ -331,53 +335,53 @@ button {
...
 </Sandpack>
 
-### Meet your first Hook {/*meet-your-first-hook*/}
+### 遇见你的第一个 Hook {/*meet-your-first-hook*/}
 
-In React, `useState`, as well as any other function starting with "`use`", is called a Hook.
+在 React 中，`useState` 以及任何其他以“`use`”开头的函数都被称为 **Hook**。
 
-*Hooks* are special functions that are only available while React is [rendering](/learn/render-and-commit#step-1-trigger-a-render) (which we'll get into in more detail on the next page). They let you "hook into" different React features.
+Hooks 是特殊的函数，只在 React [渲染](/learn/render-and-commit#step-1-trigger-a-render)时有效（我们将在下一节详细介绍）。它们能让你 “hook” 到不同的 React 特性中去。
```
Comment (daochouwangu) [inline beta/src/content/learn/state-a-components-memory.md:342]:
```suggestion
Hook 是特殊的函数，只在 React [渲染](/learn/render-and-commit#step-1-trigger-a-render)时有效（我们将在下一节详细介绍）。它们能让你 “hook” 到不同的 React 特性中去。
```

File: beta/src/content/learn/state-a-components-memory.md [hunk 9]
```diff
@@ -331,53 +335,53 @@ button {
...
 
-The only argument to `useState` is the **initial value** of your state variable. In this example, the `index`'s initial value is set to `0` with `useState(0)`. 
+`useState` 的唯一参数是 state 变量的**初始值**。在这个例子中，`index` 的初始值被`useState(0)`设置为 `0`。
 
-Every time your component renders, `useState` gives you an array containing two values:
+每次你的组件渲染时，`useState` 都会给你一个包含两个值的数组：
 
-1. The **state variable** (`index`) with the value you stored.
-2. The **state setter function** (`setIndex`) which can update the state variable and trigger React to render the component again.
+1. **state 变量** (`index`) 与你存储的值。
```
Comment (daochouwangu) [inline beta/src/content/learn/state-a-components-memory.md:368]:
```suggestion
1. **state 变量** (`index`) 会保存上次渲染的值。
```

File: beta/src/content/learn/state-a-components-memory.md [hunk 10]
```diff
@@ -887,33 +893,33 @@ button {
...
-* Use a state variable when a component needs to "remember" some information between renders.
-* State variables are declared by calling the `useState` Hook.
-* Hooks are special functions that start with `use`. They let you "hook into" React features like state.
-* Hooks might remind you of imports: they need to be called unconditionally. Calling Hooks, including `useState`, is only valid at the top level of a component or another Hook.
-* The `useState` Hook returns a pair of values: the current state and the function to update it.
-* You can have more than one state variable. Internally, React matches them up by their order.
-* State is private to the component. If you render it in two places, each copy gets its own state.
+* 当一个组件需要在多次渲染间“记住”某些信息时使用 state 变量。
+* State 变量是通过调用 `useState` Hook 来声明的。
+* Hooks 是以 `use` 开头的特殊函数。它们能让你 “hook” 到像 state 这样的 React 特性中。
```
Comment (daochouwangu) [inline beta/src/content/learn/state-a-components-memory.md:906]:
```suggestion
* Hook 是以 `use` 开头的特殊函数。它们能让你 “hook” 到像 state 这样的 React 特性中。
```

File: beta/src/content/learn/state-a-components-memory.md [hunk 11]
```diff
@@ -887,33 +893,33 @@ button {
...
+* Hooks 可能会让你想起 imports：它们需要在非条件语句中调用。调用 Hooks 时，包括 `useState`，仅在组件或另一个 Hook 的顶层被调用才有效。
```
Comment (daochouwangu) [inline beta/src/content/learn/state-a-components-memory.md:907]:
```suggestion
* Hook 可能会让你想起 import：它们需要在非条件语句中调用。调用 Hook 时，包括 `useState`，仅在组件或另一个 Hook 的顶层被调用才有效。
```

File: beta/src/content/learn/state-a-components-memory.md [hunk 12]
```diff
@@ -1270,7 +1276,7 @@ h1 { margin-top: 10px; }
 
 <Solution>
 
-First, import `useState` from React. Then replace `firstName` and `lastName` with state variables declared by calling `useState`. Finally, replace every `firstName = ...` assignment with `setFirstName(...)`, and do the same for `lastName`. Don't forget to update `handleReset` too so that the reset button works.
+首先，从 React 导入 `useState`。然后用 `useState` 声明的 state 变量替换 `firstName` 和 `lastName`。最后，用 `setFirstName(...)` 替换每个 `firstName = ...` 赋值，并对 `lastName` 做同样的事情。不要忘记更新 `handleReset`，以便重置按钮。
```
Comment (daochouwangu) [inline beta/src/content/learn/state-a-components-memory.md:1279]:
```suggestion
首先，从 React 导入 `useState`。然后用 `useState` 声明的 state 变量替换 `firstName` 和 `lastName`。最后，用 `setFirstName(...)` 替换每个 `firstName = ...` 赋值，并对 `lastName` 做同样的事情。不要忘记更新 `handleReset`，以使重置按钮生效。
```

File: beta/src/content/learn/state-a-components-memory.md [hunk 13]
```diff
@@ -1321,13 +1327,13 @@ h1 { margin-top: 10px; }
...
-#### Fix a crash {/*fix-a-crash*/}
+#### 修复一个错误 {/*fix-a-crash*/}
 
-Here is a small form that is supposed to let the user leave some feedback. When the feedback is submitted, it's supposed to display a thank-you message. However, it crashes with an error message saying "Rendered fewer hooks than expected". Can you spot the mistake and fix it?
+这是一个收集用户反馈的小表单。当反馈被提交时，它应该显示一条感谢信息。但是，现在它会发生崩溃并显示错误消息“渲染的 hooks 比预期的少”。你能发现错误并修复它吗？
 
 <Hint>
 
-Are there any limitations on _where_ Hooks may be called? Does this component break any rules? Check if there are any comments disabling the linter checks--this is where the bugs often hide!
+Hooks 只允许在*什么地方* 调用？这个组件是否违反了任何规则？检查是否有注释禁用了 linter 检查--这往往是 bug 的藏身之处!
```
Comment (daochouwangu) [inline beta/src/content/learn/state-a-components-memory.md:1336]:
```suggestion
Hook 只允许在*什么地方* 调用？这个组件是否违反了任何规则？检查是否有注释禁用了 linter 检查--这往往是 bug 的藏身之处！
```

File: beta/src/content/learn/state-a-components-memory.md [hunk 14]
```diff
@@ -1366,9 +1372,9 @@ export default function FeedbackForm() {
 
 <Solution>
 
-Hooks can only be called at the top level of the component function. Here, the first `isSent` definition follows this rule, but the `message` definition is nested in a condition.
+Hooks 只能在组件函数的顶层调用。这里，第一个 `isSent` 定义遵循这个规则，但是 `message` 的定义位于一个条件语句中。
```
Comment (daochouwangu) [inline beta/src/content/learn/state-a-components-memory.md:1375]:
```suggestion
Hook 只能在组件函数的顶层调用。这里，第一个 `isSent` 定义遵循这个规则，但是 `message` 的定义位于一个条件语句中。
```

File: beta/src/content/learn/state-a-components-memory.md [hunk 15]
```diff
@@ -1403,9 +1409,10 @@ export default function FeedbackForm() {
 
 </Sandpack>
 
-Remember, Hooks must be called unconditionally and always in the same order!
+请记住，必须在条件语句外并且始终以相同的顺序调用 Hook！
+
 
-You could also remove the unnecessary `else` branch to reduce the nesting. However, it's still important that all calls to Hooks happen *before* the first `return`.
+你还可以删除不必要的`else` 分支以减少嵌套。但要保证对 Hooks 的所有调用都发生在*第一个 `return` 前，这很重要。
```
Comment (daochouwangu) [inline beta/src/content/learn/state-a-components-memory.md:1415]:
```suggestion
你还可以删除不必要的 `else` 分支以减少嵌套。但要保证对 Hook 的所有调用都发生在*第一个 `return` 前，这很重要。
```

**Review by daochouwangu — CHANGES_REQUESTED** [review]

**Review by daochouwangu — APPROVED** [review]

**Review by daochouwangu — APPROVED** [review]

**Review by awxiaoxian2020 — APPROVED** [review]

**Pull request merged**